AOSSIE-Org/EduAid
# [Feature]  Added Audio Input for Generating Q&A

## Pull Request Title:

Feature: Add Audio Input for Generating Q&A

## Description:

### What does this PR do?
Resolves Issue #34 
This pull request adds support for audio input, allowing users to provide the path of the audio file to generate question answers directly. The feature utilizes the Facebook Wav2Vec model for audio transcription. More information about the model can be found [here](https://huggingface.co/facebook/wav2vec2-base-960h).

### Changes Made:
- Implemented audio input functionality.
- Integrated Facebook Wav2Vec model for audio transcription.
- Enhanced the Q&A generation pipeline to accommodate audio input.

### How to Test:
1. Checkout this branch locally.
2. Run the application.
3. Provide the path of an audio file.
4. Verify that the system transcribes the audio and generates accurate question answers.

### Attached Video:
[[Link to the video illustrating the working of the feature]](https://drive.google.com/file/d/16sOFmgYn20YbhSeoZUN8_Tcv9wZRtvaY/view?usp=sharing)

## Related Issues:
- #34 

## Screenshots (if applicable):
![image](https://github.com/AOSSIE-Org/EduAid/assets/102402625/f28df32d-55ea-434c-a463-d6abdaeb8669)
